apps/backend/app/auth/nextauth_auth.py

```
@@ -0,0 +1,104 @@
+        user_id = payload.get("sub") or payload.get("id")
+        
+        if not user_id:
+            # If no user ID in token, might be a session token
+            # In that case, we'd need to validate with the NextAuth API
+            logger.error("No user ID found in token")
+            raise HTTPException(status_code=401, detail="Invalid authentication token")
+
+        return user_id
+
+    except jwt.ExpiredSignatureError:
+        logger.error("Token has expired")
```

Ruff (0.14.4)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)